res/header.asm

```diff
@@ -306,7 +320,18 @@
 ; Desc  - Prints integer on top of stack to STDOUT
 ; -----------------------------------------------------------------------------
 %macro O_PRTI 0
-  ; TBD
+  POP   RAX                 ; Gets the integer to be printed from the stack
```

Comment:
This will not print the integer. You need to progressively divide the integer by 10 & get the remainder, convert it to ASCII and print it. Here is some starter code that you can refine:

```
    ; assume number is in eax
    mov ecx, 10

loophere:
    mov edx, 0
    div ecx

    ; now eax <-- eax/10
    ;     edx <-- eax % 10

    ; print edx
    ; this is one digit, which we have to convert to ASCII
    ; the print routine uses edx and eax, so let's push eax
    ; onto the stack. we clear edx at the beginning of the
    ; loop anyway, so we don't care if we much around with it

    push eax

    ; convert dl to ascii
    add dl, '0'

    mov ah,2  ; 2 is the function number of output char in the DOS Services.
    int 21h    ; calls DOS Services

    ; now restore eax
    pop eax

    ; if eax is zero, we can quit

    cmp eax, 0
    jnz loophere
```

Reply:
Convert to 64-bit & linux system calls instead of DOS